lib/src/play_store_search_api.dart

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021 William Kwabla. All rights reserved.
+ */
+
+import 'package:html/dom.dart';
+import 'package:html/parser.dart' show parse;
+import 'package:http/http.dart' as http;
+
+class PlayStoreSearchAPI {
+  /// Play Store Search Api URL
+  final String playStorePrefixURL = 'play.google.com';
+
+  /// Provide an HTTP Client that can be replaced for mock testing.
+  http.Client? client = http.Client();
+
+  bool debugEnabled = false;
+
+  /// Look up by id.
+  Future<Document?> lookupById(String id) async {
+    if (id.isEmpty) {
+      return null;
+    }
+
+    final url = lookupURLById(id)!;
+
+    final response = await http.get(Uri.parse(url));
+
+    if (response.statusCode != 200) {
+      print('Can\'t find an app in the Play Store with the id: $id');
+      return null;
+    }
+
+    final decodedResults = _decodeResults(response.body);
+
+    return decodedResults;
+  }
+
+  String? lookupURLById(id) {
+    final url =
+        Uri.https(playStorePrefixURL, '/store/apps/details', {'id': '$id'})
+            .toString();
+
+    return url;
+  }
+
+  Document? _decodeResults(jsonResponse) {
+    if (jsonResponse.isNotEmpty) {
+      final decodedResults = parse(jsonResponse);
+
+      return decodedResults;
+    }
+    return null;
+  }
+}
+
+class PlayStoreResults {
+  /// Return field releaseNotes from Play Store results.
+  static String? releaseNotes(response) {
+    final sectionElements = response.getElementsByClassName('W4P4ne');
+    final releaseNotesElement = sectionElements.firstWhere(
+      (elm) => elm.querySelector('.wSaTQd')!.text == 'What\'s New',
+    );
+    final releaseNotes = releaseNotesElement
+        .querySelector('.PHBdkd')
+        ?.querySelector('.DWPxHb')
+        ?.text;
+
+    return releaseNotes;
+  }
+
+  /// Return field trackViewUrl from Play Store results.
+  static String? trackViewUrl(id) {
```

Comment:
@Williano This method `trackViewUrl` does not return anything from the Play Store results. What is this for? It is no different from `lookupURLById`.

Reply:
Yes, is not different from lookupURLById